flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -1061,8 +1064,12 @@ private static String getDefaultName() {
 	 * @return The execution environment of the context in which the program is executed.
 	 */
 	public static ExecutionEnvironment getExecutionEnvironment() {
+
 		return contextEnvironmentFactory == null ?
-				createLocalEnvironment() : contextEnvironmentFactory.createExecutionEnvironment();
+				(contextEnvironmentFactoryThreadLocal.get() == null ?
```

Comment:
Do we need to add `contextEnvironmentFactoryThreadLocal` check in the function `areExplicitEnvironmentsAllowed`(line 1284). I can't add a review there.

Reply:
Thanks for your review @klion26. Good catch and updated.